sections/docker.zsh

```diff
@@ -24,13 +24,33 @@ spaceship_docker() {
 
   spaceship::exists docker || return
 
+  # Better support for docker environment vars: https://docs.docker.com/compose/reference/envvars/
+  # Use COMPOSE_PATH_SEPARATOR or colon as default
+  local separator=":"
+  [[ ! -z $COMPOSE_PATH_SEPARATOR ]] && separator=$COMPOSE_PATH_SEPARATOR
+
+  # COMPOSE_FILE may have several filenames separated by colon, test all of them
+  local filenames=("${(@ps/$separator/)COMPOSE_FILE}")
+
+  local compose_exists
+  for filename in $filenames; do
+    if [[ ! -f $filename ]]; then
+      compose_exists=false
+      break
+    fi
+    compose_exists=true
+  done
```

Comment:
This iterates over every elements whatever the case, We could break the loop when we encounter first file.

```diff
-  local compose_exists
+  local compose_exists=false
   for filename in $filenames; do
-    if [[ ! -f $filename ]]; then
-      compose_exists=false
+    if [[ -f $filename ]]; then
+      compose_exists=true
       break
     fi
-    compose_exists=true
```

Reply:
I'm not sure about that... for instance:
`COMPOSE_FILE=docker-compose-yml:services.yml` must match only if both files exist... testing only one, `docker-compose.yml` would match at several another directories.

Reply:
If there's a file named `docker-compose.yml` in current directory, We could display the `docker` section. Why should we check further ?

Reply:
That's my real case: 
`COMPOSE_FILE=infra/base-services.yml:infra/services.yml:infra/docker-compose.yml:infra/docker-compose.override.yml`

Even with `docker-compose.yml` available in current directory, try a `docker-compose up -d` raise an error:
`ERROR: .IOError: [Errno 2] No such file or directory: u'./infra/base-services.yml'`

So, if `COMPOSE_FILE` is present, docker-compose.yml isn't sufficient to show docker prompt.

I'll fix the code for that case!

Reply:
So, If I've `COMPOSE_FILE=docker-compose.yml:docker-compose.prod.yml` and the current directory only have `docker-compose.prod.yml` and not `docker-compose.yml`. Shouldn't we display the `docker` section ?

![docker compose](https://shipusercontent.com/3c79960e252dae67c64e9e516124b3b3/Pasted%20Image%201.png)

Reply:
No, we shouldn't. 'Cause the command `docker-compose up -d` will fail!

Reply:
If that's the intended behavior, then great 👍

Reply:
Maybe the best approach (but requires a lot of more effort) is split into two sections:
docker (check for Dockerfile) and docker-compose (check for $COMPOSE_FILE or docker-compose.yml).

For now, I've already done that fix.